lib/demo_mode/persona.rb

```diff
@@ -31,9 +31,16 @@ def begin_demo(&block)
       if block
         @begin_demo = block
       else
-        @begin_demo || proc do
-          sign_in @session.signinable
-          redirect_to main_app.root_path
+        @begin_demo || proc do |option|
+          proc do
```

Comment:
Can you explain what the proc nesting is doing here?

Reply:
If we are not returning a proc, [this](https://github.com/Betterment/demo_mode/pull/59/files#diff-7061eec77ccb3a1217893822312e65f7c45fde9996183101f9abfb0f3e7cc4a4R48) won't evaluate it correctly.